File: examples/python-api/app/application/user/dtos.py
"""User Application DTOs."""

from dataclasses import dataclass

from app.domain.user.entity import User


@dataclass(frozen=True)
class CreateUserDto:
    email: str
    name: str


@dataclass(frozen=True)
class UserResponseDto:
    id: str
    email: str
    name: str
    status: str

    @classmethod
    def from_entity(cls, user: User) -> "UserResponseDto":
        return cls(
            id=user.id,
            email=user.email,
            name=user.name,
            status=user.status,
        )

File: examples/typescript-api/src/application/user/dtos.ts
/**
 * Application DTOs
 */

export class CreateUserDto {
  constructor(
    public readonly email: string,
    public readonly name: string,
  ) {}
}

export class UserResponseDto {
  constructor(
    public readonly id: string,
    public readonly email: string,
    public readonly name: string,
    public readonly status: string,
  ) {}

  static from(user: { id: string; email: { value: string }; name: string; status: string }) {
    return new UserResponseDto(user.id, user.email.value, user.name, user.status);
  }
}
